scripts/update-opam-in-docker.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# update OPAM packages, including pinned ones
+
+if [ -z $CODA_DOCKER ] ; then
+    echo `basename $0` "can run only inside Coda Docker image"
+    exit 1
+fi
+
+# update not-pinned OPAM packages
+
+opam update
+
+opam switch import src/opam.export
+
+# update pinned packages
+
+git submodule update --init --recursive
+
+# reinstalls of pinned packages should not use saved sources
+declare -x OPAMKEEPBUILDDIR=false
+declare -x OPAMREUSEBUILDDIR=false
+
+function reinstall () {
+    echo "Updating OPAM package $1"
+    opam reinstall -y $1
+}
+
+function uptodate () {
+    echo "OPAM package $1 is up-to-date"
+}
+
+# submodules
+for pkg in async_kernel digestif graphql_ppx ocaml-extlib ppx_optcomp rpc_parallel ; do
+    CURRENT_COMMIT=$(git submodule status src/external/$pkg | awk '{print $1}')
+    DOCKER_COMMIT=$(cat ~opam/opam-repository/$pkg.commit)
+    if [ $CURRENT_COMMIT != $DOCKER_COMMIT ] ; then
```

Comment:
Could we make this check conditional on `CODA_DOCKER` being true? This would be super useful to mention in `CONTRIBUTING.md`/`README-dev.md` for first set-up.